orttraining/orttraining/training_api/module.cc

```diff
@@ -20,6 +24,116 @@ namespace {
 // TODO: consolidate with frontend tooling
 const std::string ACCUMULATE_GRAD_CONTROL_INPUT_NAME{"lazy_reset_grad"};
 
+std::unordered_set<const Node*> GetReverseReachableNodes(Graph& inference_graph,
```

Comment:
is this existing api helpful? 

  // Reversely traverse to get reachable nodes.
  graph_.ReverseDFSFrom(
      nodes, {}, [&reachable_nodes](const Node* n) { reachable_nodes.insert(n->Index()); });